### PR TITLE
Version bump to atari800 4.2.0.

### PR DESCRIPTION
**Pull request by krischik**
#### Description

* Update Version.
* Fix startup scripts:
   * Add video acceleration.
   * Use new window size option.
   * Various other startup options.
   * Use separate configuration files.
   * Save configuration on exit.
* Fix file modes on installation

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.6
Xcode  12.0

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by kencu** [comment]
Thanks! Looks good, and builds widely. We'll commit soon. There are a couple of minor style quibbles we'll fix up to match our usual way; please don't be frustrated with us over those.

**Comment by kencu** [comment]
re: the commit message. We have some written guidelines somewhere about this, and I don't immediately know where those are, but the TL;DR is the first line of your commit message should read something like this:
```
atari800: bump to 4.2.0
```
having them all in that pattern makes the long line of commits much easier to read, when you're reading hundreds of them, as we do

File: emulators/atari800/Portfile [hunk 1]
```diff
@@ -3,26 +3,31 @@
...
 
 name                        atari800
-version                     3.1.0
-revision                    2
+version                     4.2.0
+revision                    0
 platforms                   darwin
 categories                  emulators
-maintainers                 nomaintainer
+maintainers                 krischik, openmaintainer
```
[the remark this answers is not in the document]
Comment (krischik) [reply emulators/atari800/Portfile:10]: I haven't done a lot recently but krischik is indeed both my MacPorts and my GitHub handle:

https://trac.macports.org/wiki/krischik
https://github.com/krischik

So is it:

```
maintainers         {krischik @krischik} openmaintainer
```

[the remark this answers is not in the document]
Comment (mf2k) [reply emulators/atari800/Portfile:10]: @krischik: Yes, exactly.

File: emulators/atari800/Portfile [hunk 2]
```diff
@@ -3,26 +3,31 @@
...
+set tagversion              ATARI800_4_2_0
+set patchversion            4.2.0
 
-set patchversion            52737
-set prog                    atari800-${version}.tar.gz
-set rom                     xf25.zip
+homepage                    https://atari800.github.io
+master_sites                git:https://github.com/atari800/atari800/releases/download/${tagversion}:prog                                  \
+                            git:https://github.com/krischik/macports-user-krischik/releases/download/atari800-${patchversion}:appbundles   \
+                            http://www.emulators.com/freefile:pcxf
```
[the remark this answers is not in the document]
Comment (krischik) [reply emulators/atari800/Portfile:20]: Now that you mention it: The rom is still on SourceForge.

**Comment by krischik** [comment]
Fixed everything from the code review. Git was the most annoying part ;-)

**Comment by kencu** [comment]
OK, thanks. This looks sufficiently clean to me to commt... but what's happened to github? There are a bunch of other commits in this branch now, it's barfing on us! I can just take your commit as a patch and do it that way to my macportsofficial/macports-ports checkout, but do you see how to fix up this commit branch so I can just rebase and merge it?

Perhaps you might know just what commands you did to get here.

Usually, on a branch that I'm working on, I just make a few changes if needed, run an interactive rebase to smush them together, and force-push to the branch. I'm guessing that perhaps you rebased your branch on master? I'm sorry, my git-fu is not strong at this moment to tell you exactly what to do to make this right.

**Comment by krischik** [comment]
@kencu Those are the commits of the last two days from **macports/macports-ports** — I'll regularly merge **macports/macports-ports** to **krischik/macports-ports** to keep my fork up to date. 

Those commits already exist **macports/macports-ports** and as far as I see it there is nothing wrong with the pull request.

Do note that I'm not getting payed for this so there is a limit on how much I'm willing to do. I already spend almost as much time fighting git then updating the Portfile.

**Comment by kencu** [comment]
committed directly as I didn't know how to concisely tell you how to fix this, and time to move on.

Thanks for taking over maintainership! Good luck!

**Comment by krischik** [comment]
More luck with git with the next time a new version comes out.

**Comment by kencu** [comment]
Your error was merging changes in master into your branch. You can't do it the way you were doing it.

Sort it out next time (or not!).

**Comment by krischik** [comment]
I'll guess I avoid updating the fork as long as a pull request us open.

Even true it's standard procedure elsewhere.

**Comment by kencu** [comment]
you have to do it a certain way. I don't do it often, but I believe you update the master branch then "rebase" your feature branch on master. 

MacPorts doesn't do things any differently than anywhere else AFAIK.

It takes a while to get used to git.

**Pull request closed without merging**